SPHINXsys/src/shared/particle_dynamics/solid_dynamics/thin_structure_math.cpp

```diff
@@ -243,5 +243,30 @@ namespace SPH
 			return corrected_almansi_strain;
 		}
 		//=================================================================================================//
+		Mat2d getCorrectionMatrix(const Mat2d &local_deformation_part_one)
+		{
+			Mat2d correction_matrix(0.0);
+			correction_matrix[0][0] = 1.0 / (local_deformation_part_one[0][0] + TinyReal);
+			return correction_matrix;
+		}
+		//=================================================================================================//
+		Mat3d getCorrectionMatrix(const Mat3d &local_deformation_part_one)
+		{
+			Mat2d reduced_local_deformation_part_one;
+			reduced_local_deformation_part_one[0][0] = local_deformation_part_one[0][0];
+			reduced_local_deformation_part_one[0][1] = local_deformation_part_one[0][1];
+			reduced_local_deformation_part_one[1][0] = local_deformation_part_one[1][0];
+			reduced_local_deformation_part_one[1][1] = local_deformation_part_one[1][1];
```

Comment:
+TinyReal on the diagonal ?

Reply:
From what I have tested, using 0 is fine.

Reply:
It would be preferable to have it uniform, so either removing it from 2D version or add it to the 3D

Reply:
Yes, you are right. I have removed TinyReal from the 2D model as we have initialized global_deformation with Eps.